### PR TITLE
feat: workflow validation logic and constraints (task-120) [v2]

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

Implements `WorkflowValidator` class for semantic validation of workflow configurations:

- **State validation**: Ensures required `To Do` initial state exists, validates terminal states (Done, Deployed, Cancelled, Archived)
- **Graph validation**: Detects cycles using DFS, validates reachability from initial state using BFS
- **Reference validation**: Checks all state references in workflows/transitions point to defined states
- **Agent validation**: Warns on unknown agent names in workflow configurations
- **Convenience function**: `validate_workflow(config)` for one-liner validation

### v2 Changes (PR #88 feedback)
- Added `validate_workflow` to module exports in `__init__.py`
- Fixed `to_state` validation in `_check_cycles` (line 467)
- Fixed `to_state` validation in `_check_reachability` (line 531)
- Fixed `to_state` validation in `_check_terminal_states` (line 587)

All three graph building blocks now verify `to_state in self._states`, not just truthy checks.

## Test plan
- [x] 57 unit tests covering all validation scenarios
- [x] Tests for cycle detection (simple, self-loop, longer cycles)
- [x] Tests for reachability analysis (single and multiple unreachable states)
- [x] Tests for undefined state/workflow references
- [x] Tests for edge cases (empty/invalid config formats)
- [x] Ruff linting and formatting pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/specify_cli/workflow/__init__.py [hunk 1]
```diff
@@ -1,30 +1,17 @@
...
-    "WorkflowConfigNotFoundError",
-    "WorkflowConfigValidationError",
-    "WorkflowNotFoundError",
-    "WorkflowStateError",
+    "ValidationSeverity",
+    "ValidationIssue",
+    "ValidationResult",
+    "WorkflowValidator",
+    "validate_workflow",
 ]
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/workflow/__init__.py]: fix me

File: src/specify_cli/workflow/validator.py [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+        # Extract and normalize states to a set
+        states_raw = config_data.get("states", [])
+        if isinstance(states_raw, list):
+            # Handle both simple strings and state objects with 'name' key
+            self._states: set[str] = set()
+            for state in states_raw:
+                if isinstance(state, str):
+                    self._states.add(state)
+                elif isinstance(state, dict) and "name" in state:
+                    self._states.add(state["name"])
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/workflow/validator.py]: fix

**Comment by jpoley** [comment]
make fixes and resubmit
